File: drift/src/main/java/drift/com/drift/helpers/TextHelper.java
package drift.com.drift.helpers;


import android.text.Editable;
import android.text.Html;
import android.text.SpannableStringBuilder;
import android.text.util.Linkify;

public class TextHelper {

    public static String cleanString(String body){
        body = body.replaceFirst("\\s++$", ""); //Trim trailing white spaces
        body = body.replaceAll("<p dir=\"....?\">", "");
        body = body.replaceAll("<p>", "");
        body = body.replaceAll("</p>", "<br />");

        body = body.replaceFirst("<p dir=\"....?\">", "");
        body = body.replaceFirst("<p>", "");

        if (body.endsWith("</p>")) {
            body = body.substring(0, body.lastIndexOf("</p>"));
        }

        return body;
    }

    public static String wrapTextForSending(String body) {

        Editable editableText = new SpannableStringBuilder( body );
        Linkify.addLinks(editableText, Linkify.WEB_URLS);
        body = Html.toHtml(editableText).trim();

        body = body.replaceAll("\n", "<br />");
        return body;
    }

}

File: drift/src/main/java/drift/com/drift/helpers/TextHelper.kt
package drift.com.drift.helpers


import android.text.Html
import android.text.SpannableStringBuilder
import android.text.util.Linkify

object TextHelper {

    fun cleanString(body: String): String {
        var body = body
        body = body.replaceFirst("\\s++$".toRegex(), "") //Trim trailing white spaces
        body = body.replace("<p dir=\"....?\">".toRegex(), "")
        body = body.replace("<p>".toRegex(), "")
        body = body.replace("</p>".toRegex(), "<br />")

        body = body.replaceFirst("<p dir=\"....?\">".toRegex(), "")
        body = body.replaceFirst("<p>".toRegex(), "")

        if (body.endsWith("</p>")) {
            body = body.substring(0, body.lastIndexOf("</p>"))
        }

        return body
    }

    fun wrapTextForSending(body: String): String {
        var body = body

        val editableText = SpannableStringBuilder(body)
        Linkify.addLinks(editableText, Linkify.WEB_URLS)
        body = Html.toHtml(editableText).trim { it <= ' ' }

        body = body.replace("\n".toRegex(), "<br />")
        return body
    }

}
